src/seedu/addressbook/AddressBook.java

```diff
@@ -202,13 +202,25 @@ public static void main(String[] args) {
         showWelcomeMessage();
         processProgramArgs(args);
         loadDataFromStorage();
-        while (true) {
-            String userCommand = getUserInput();
-            echoUserCommand(userCommand);
-            String feedback = executeCommand(userCommand);
-            showResultToUser(feedback);
-        }
+        readAndProcessAllInput();
     }
+    
+	private static void readAndProcessAllInput() {
```

Comment:
Always try to comment your functions. It may seem trivial but it makes your intent much clearer.